### PR TITLE
[UXE-4804] fix: error when using advanced filters

### PR DESCRIPTION
**Pull request by pauloSF0**
### Explain what was fixed and the correct behavior.
 Error when using advanced filters: Numeric URIs interpreted as dates in GQL
### Does this PR introduce UI changes? Add a video or screenshots here.

https://github.com/user-attachments/assets/40bdf4fa-92bb-4ee5-bf6a-e2d52b038e4a

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] bug: TITLE
- [x] Commits are tagged with the right word (feat, test, refactor, etc)
- [x] Application responsiveness was tested to different screen sizes
- [ ] New tests are added to prevent the same issue from happening again
- [ ] UI changes are validated by a team designer
- [x] Code is formatted and linted
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

**Review by lucasmendes21 — DISMISSED** [review]

**Comment by vinigfer** [comment]
Is it possible to add a unit test for this fix?

File: src/helpers/convert-gql.js [hunk 1]
```diff
@@ -16,14 +16,37 @@ const getGraphQLType = (value) => {
...
-    if (!isNaN(Date.parse(value))) {
+    if (isValidDate(value)) {
       return 'DateTime'
     }
 
     return 'String'
   }
 }
 
+function isValidDate(dateString) {
```
Comment (peterpaulo-azion) [inline src/helpers/convert-gql.js:27]: Algo assim não resolveria? No caso, ela verificaria se a string é uma data válida ou não de uma forma mais limpa.

```suggestion
function isValidDate(dateString) {
  const date = new Date(dateString);
  return date instanceof Date && !isNaN(date.getTime());
}
```

File: src/helpers/convert-gql.js [hunk 2]
```diff
@@ -138,9 +161,12 @@ const separateFieldsByType = (fields) => {
 const mergeFieldsIntoFilter = (fields, filter) => {
   fields.forEach(({ operator, valueField, value }) => {
     const filterKey = operator === 'In' ? 'in' : 'and'
+    const isOperetorContains = operator === 'Like'
+
+    const fieldValue = isOperetorContains ? `%${value}%` : value
```
Comment (vinigfer) [inline src/helpers/convert-gql.js:166]: Typo
```suggestion
    const fieldValue = isOperatorContains ? `%${value}%` : value
```

File: src/helpers/convert-gql.js (hunk 1 above)
Comment (pauloSF0) [reply src/helpers/convert-gql.js:27]: Não, isso ai não funciona

**Review by peterpaulo-azion — APPROVED** [review]

File: src/helpers/convert-gql.js [hunk 3]
```diff
@@ -16,14 +16,33 @@ const getGraphQLType = (value) => {
...
+    if (isValidDate(value)) {
       return 'DateTime'
     }
 
     return 'String'
   }
 }
 
+function isValidDate(dateString) {
+  const dateRegexes = [
```
Comment (william-tome) [inline src/helpers/convert-gql.js:28]: Testa essa lib, conforme o @lucasmendes21 sugeriu. Me parece que tem outros pontos no projeto que ela seria aproveitada.

https://date-fns.org/v4.1.0/docs/isDate

File: src/helpers/convert-gql.js [hunk 4]
```diff
@@ -49,6 +68,15 @@ function buildGraphQLQuery({ filterParameter, dataset, limit, orderBy, filterQue
   ].join('\n')
 }
 
+const formatValueContainOperator = (variable) => {
+  for (const key in variable) {
+    if (Object.prototype.hasOwnProperty.call(variable, key) && key.includes('Like')) {
```
Comment (william-tome) [inline src/helpers/convert-gql.js:73]:
```suggestion
    if (variable[key] && key.includes('Like')) {
```

**Review by william-tome — COMMENTED** [review]
Melhorias

**Review by william-tome — APPROVED** [review]

**Review by lucasmendes21 — APPROVED** [review]

**Pull request merged**